src/sql/workbench/contrib/notebook/find/notebookFindModel.ts

```diff
@@ -523,17 +523,32 @@ export class NotebookFindModel extends Disposable implements INotebookFindModel
 		return this.findArray;
 	}
 
-	private searchFn(cell: ICellModel, exp: string, maxMatches?: number): NotebookRange[] {
+	private searchFn(cell: ICellModel, exp: string, matchCase: boolean = false, wholeWord: boolean = false, maxMatches?: number): NotebookRange[] {
 		let findResults: NotebookRange[] = [];
 		let cellVal = cell.cellType === 'markdown' ? this.cleanUpCellSource(cell.source) : cell.source;
 		let index: number;
 		let start: number;
 		let end: number;
+		let wholeWordRegex = new RegExp(`\\b${exp}\\b`);
 		if (cellVal) {
+
 			if (typeof cellVal === 'string') {
 				index = 0;
-				while (cellVal.substr(index).toLocaleLowerCase().indexOf(exp.toLocaleLowerCase()) > -1) {
-					start = cellVal.substr(index).toLocaleLowerCase().indexOf(exp.toLocaleLowerCase()) + index;
+				let cellValFormatted = cellVal;
+				if (!matchCase) {
+					cellValFormatted = cellVal.toLocaleLowerCase();
+					exp = exp.toLocaleLowerCase();
+					wholeWordRegex = new RegExp(`\\b${exp}\\b`);
+				}
+				while (cellValFormatted.substr(index).indexOf(exp) > -1) {
```

Comment:
Since we aren't storing the substring anywhere, is there an alternate contains/search method we can use here instead? It would avoid creating extra strings.

Reply:
They might have indexing options to limit the search to a certain range.

Reply:
Actually, we make 3 calls for substring here, so we can probably just do that once and re-use the value elsewhere in this loop.

Reply:
Like calling substring as part of the iterate step of a for loop, for example.